src/realm/replication.cpp

```diff
@@ -109,34 +400,57 @@ void Replication::list_clear(const CollectionBase& list)
 {
     select_collection(list);           // Throws
     m_encoder.collection_clear(list.size()); // Throws
+    if (auto logger = get_logger()) {
+        logger->log(util::Logger::Level::trace, "   Clear '%1'", get_prop_name(list.get_short_path()));
+    }
 }
 
 void Replication::link_list_nullify(const Lst<ObjKey>& list, size_t link_ndx)
 {
     select_collection(list);
     m_encoder.collection_erase(link_ndx);
+    if (auto logger = get_logger()) {
+        logger->log(util::Logger::Level::trace, "   Nullify '%1' position %2",
+                    m_selected_table->get_column_name(list.get_col_key()), link_ndx);
+    }
 }
 
-void Replication::dictionary_insert(const CollectionBase& dict, size_t ndx, Mixed, Mixed)
+void Replication::dictionary_insert(const CollectionBase& dict, size_t ndx, Mixed key, Mixed value)
 {
     select_collection(dict);
     m_encoder.collection_insert(ndx);
+    if (auto logger = get_logger()) {
+        if (logger->would_log(util::Logger::Level::trace)) {
```

Comment:
should wrap everything in macro (including logger check and logging level) otherwise a lot of unneeded calls.

Reply:
I am allergic to macros. And a macro will not remove the calls from the program (if I got your suggestion right).